src/components/modal/add-restaurant-modal.ts

```diff
@@ -0,0 +1,91 @@
+import $form from "../form-elements/form.ts";
+import $restaurantItem from "../restaurant/restaurant-item.ts";
+import { currentRestaurantData } from "../../data/restaurant.ts";
+import { CATEGORY_ICON } from "../../constants/iconPath.ts";
+import { saveRestaurantsToLocalStorage } from "../../data/restaurant.ts";
+
+export type RestaurantProps = {
+  category: string;
+  name: string;
+  distance: number;
+  description: string;
+  link: string;
+  isFavorite: boolean;
+};
+
+type ModalProps = {
+  form: HTMLElement | HTMLElement[];
+};
+
+export const addRestaurant = (data: RestaurantProps): void => {
+  handleModalClose();
+  const lastId =
+    currentRestaurantData.length > 0
+      ? Math.max(...currentRestaurantData.map((r) => r.dataId))
+      : 0;
+  const newId = lastId + 1;
+  const categoryIcon = CATEGORY_ICON[data.category];
+  const newRestaurant = {
```

Comment:
정의해둔 type을 이용하면 타입시스템의 도움을 받을 수 있을거 같아요

```suggestion
  const newRestaurant: Restaurant = {
```

Reply:
https://github.com/woowacourse/javascript-lunch/pull/241/commits/cd61de027baabc00c9d6a40f81d7bd4a06f9cf6d 반영했습니다!